src/Http/Routing/src/EndpointRoutingMiddleware.cs

```diff
@@ -102,6 +108,35 @@ private Task SetRoutingAndContinue(HttpContext httpContext)
             }
 
             Log.MatchSuccess(_logger, endpoint);
+
+            var shortCircuitMetadata = endpoint.Metadata.GetMetadata<ShortCircuitMetadata>();
+            if (shortCircuitMetadata is not null)
+            {
+                if (!_routeOptions.SuppressCheckForUnhandledSecurityMetadata)
+                {
+                    if (endpoint.Metadata.GetMetadata<IAuthorizeData>() is not null)
+                    {
+                        ThrowMissingAuthMiddlewareException(endpoint);
+                    }
+
+                    if (endpoint.Metadata.GetMetadata<ICorsMetadata>() is not null)
+                    {
+                        ThrowMissingCorsMiddlewareException(endpoint);
+                    }
+                }
+
+                if (shortCircuitMetadata.StatusCode.HasValue)
+                {
+                    httpContext.Response.StatusCode = shortCircuitMetadata.StatusCode.Value;
+                }
+
+                if (endpoint.RequestDelegate is not null)
+                {
+                    return endpoint.RequestDelegate(httpContext);
+                }
+
+                return Task.CompletedTask;
+            }
```

Comment:
I think this part can potentially be extracted to a separate middleware; it does not need to be handled inside routing directly.

Typically, routing is only responsible for the decision of what to execute, not when.

You can have a separate middleware in the pipeline to decide exactly when to eagerly execute the endpoint exactly where you want it, which gives you more flexibility and doesn't make this a special concern of the routing middleware.

Reply:
@javiercn The integration with routing was deliberate. I'll follow up with you directly.